include/graaflib/types.h

```diff
@@ -9,14 +9,20 @@ namespace graaf {
 using vertex_id_t = std::size_t;
 using edge_id_t = std::pair<vertex_id_t, vertex_id_t>;
 
+template <class T>
+inline void hash_combine(std::size_t& seed, const T& v) {
```

Comment:
Adding the Boost license:

```suggestion

/**
 * Based on boost::hash_combine. Since Boost is licensed under the Boost
 * Software License, we include a copy of the license here.
 * TODO(b.luppes): consider using Boost's new hash_combine implementation
 *
 * Boost Software License - Version 1.0 - August 17th, 2003
 *
 * Permission is hereby granted, free of charge, to any person or organization
 * obtaining a copy of the software and accompanying documentation covered by
 * this license (the "Software") to use, reproduce, display, distribute,
 * execute, and transmit the Software, and to prepare derivative works of the
 * Software, and to permit third-parties to whom the Software is furnished to
 * do so, all subject to the following:
 *
 * The copyright notices in the Software and this entire statement, including
 * the above license grant, this restriction and the following disclaimer,
 * must be included in all copies of the Software, in whole or in part, and
 * all derivative works of the Software, unless such copies or derivative
 * works are solely in the form of machine-executable object code generated by
 * a source language processor.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE, TITLE AND NON-INFRINGEMENT. IN NO EVENT
 * SHALL THE COPYRIGHT HOLDERS OR ANYONE DISTRIBUTING THE SOFTWARE BE LIABLE
 * FOR ANY DAMAGES OR OTHER LIABILITY, WHETHER IN CONTRACT, TORT OR OTHERWISE,
 * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 * DEALINGS IN THE SOFTWARE.
 *
 */
template <class T>
inline void hash_combine(std::size_t& seed, const T& v) {
```

Reply:
Ah, apparently I cannot push changes nor can I accept code suggestions on your fork @laurence-atomic.
Could you please accept/add the license if it looks good to you?

Reply:
I agree that using the latest version of the Boost hash function will probably improve performance even more. This update is fast enough for our use-case though. I'll accept the change request and push it back to you.